notNeededPackages.json

```diff
@@ -1253,6 +1253,10 @@
             "libraryName": "google-maps-react",
             "asOfVersion": "2.0.5"
         },
+        "googlemaps": {
+            "libraryName": "@types/google.maps",
+            "asOfVersion": "3.44.0"
+        },
```

Comment:
There was some discussion in previous PRs about this. Someone from Google was saying that they were planning to release types that are auto-generated from source. I assume that's what `@types/google.maps` is now, and the bot probably updates those types every time they release a new Maps JavaScript API version.

So yeah https://www.npmjs.com/package/googlemaps is not shipping its own types, that package is not even from Google. There's not really an official package that the types could be shipped with, which is probably why they are generated into this repo. The official package would be [`@googlemaps/js-api-loader`](https://www.npmjs.com/package/@googlemaps/js-api-loader) but that doesn't have the actual API code in it afaik, just a function to load it (a lot of people would load it via script tag though).

The deprecation should say to use `@types/google.maps` instead of these manually written types. I think that's what the author was trying to achieve (at least judging by the title).

Reply:
@jpoehnelt is the guy from Google, and just found the issue via [search](https://github.com/DefinitelyTyped/DefinitelyTyped/issues?q=author%3Ajpoehnelt) where he asks about type auto-generation: #49210.

Reply:
@andrewbranch I figured that's the case, but it's the best thing I can do as a contributor. If the deprecation message could manually be set to redirect users to the correct package, that would obviously be a lot better.

The relation between the packages are somewhat confusing to me too. I know that there was a DefinitelyTyped package for the google maps API that was called `googlemaps`, and multiple other DefinetelyTyped packages depends on these types. But since #49873, there is now a type that is synced from the google maps codebase which should be more correct and up to date. It was then renamed (#51154) to `google.maps`. 

So what I'd like to signify is that the package `@types/googlemaps` is deprecated and that users should install `@types/google.maps` instead, but the only option I really have as a contributor is to use the scripts for removing a package. If a maintainer could do some more magic for the package in question, that'd be great.

Reply:
Well, I guess my main question is, does `@types/googlemaps` actually correspond to https://www.npmjs.com/package/googlemaps (whether intentionally or accidentally)? The name match implies that it does, and the implementation library, despite being four years stale, is not itself deprecated. So the problem is that if someone is using `googlemaps` from npm, `@types/googlemaps` may still be the correct set of type definitions for that very outdated implementation library. Although we would probably like that person to stop using `googlemaps` from npm altogether, it would be incorrect to deprecate `@types/googlemaps`—the old types correctly correspond to the old library. We only deprecate `@types` packages to signify that users of a given implementation library should now get types _for that implementation library_ elsewhere. And since `googlemaps` is no longer being updated, `@types/google.maps` is probably not where `googlemaps` users should get their typings. Does that make sense?

On the other hand, if `@types/googlemaps` actually does not correspond to `googlemaps` and the name overlap is just an unfortunate coincidence, and `googlemaps` users actually _don’t_ have any typings that correspond to their library, then I can issue a custom one-off deprecation message for `@types/googlemaps`.

Reply:
That's a good and well formulated question. I was honestly not aware of the `googlemaps` package, but I've tried to look into at it, and I'd say the naming colision is an unfortunate accident more than anything else. The `googlemaps` package is a Node package that exports a [`GoogleMapsAPI` method](https://github.com/moshen/node-googlemaps/blob/master/lib/index.js#L46) that can be used to load a Google Maps API into node and preform various operations that the Javascript library allows. While `@types/googlemaps` is the DefinitelyTyped typings of the Google Maps Web API itself, which defines the appropiate API methods on the global [`google.maps`.](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/googlemaps/index.d.ts#L79). 

The versioning of `@types/googlemaps` also follows the [Google Maps web API versioning](https://developers.google.com/maps/documentation/javascript/versions#choosing-a-version-number), rather than the versioning of `googlemaps`, the latest version released of that library is `1.12.0`.

Reply:
Thanks for checking that out—I think you’re right; the types don’t seem to match the library at all. Please go ahead and remove your change to `notNeededPackages.json`, and I’ll discuss with the team about manually issuing a custom deprecation message for `@types/googlemaps`. Thanks!

Reply:
Of course! Done! @andrewbranch

Reply:
You are right, these types are for the Google Maps API, not for any NPM package. The naming is unfortunate and confusing for everyone, so it's good to see progress being made in cleaning things up a bit.